skore/tests/unit/sklearn/estimator/feature_importance/test_mean_decrease_impurity.py

```diff
@@ -0,0 +1,218 @@
+                n_informative=3,
+                random_state=42,
+            ),
+            make_pipeline(StandardScaler(), RandomForestClassifier(random_state=0)),
+            (5, 1),
+        ),
+        (
+            make_classification(n_features=5, random_state=42),
+            make_pipeline(StandardScaler(), RandomForestClassifier(random_state=0)),
+            (5, 1),
+        ),
+        (
+            make_regression(n_features=5, n_targets=3, random_state=42),
+            RandomForestRegressor(random_state=0),
+            (5, 1),
+        ),
+    ],
+)
+def test_numpy_arrays(data, estimator, expected_shape):
+    X, y = data
+    estimator.fit(X, y)
+    report = EstimatorReport(estimator)
+    result = report.feature_importance.mean_decrease_impurity()
+
+    assert result.shape == expected_shape
+
+    expected_index = (
+        [f"x{i}" for i in range(X.shape[1])]
+        if isinstance(estimator, Pipeline)
+        else [f"Feature #{i}" for i in range(X.shape[1])]
+    )
+    assert result.index.tolist() == expected_index
+
+    expected_columns = ["Mean decrease impurity"]
+    assert result.columns.tolist() == expected_columns
+
+
+@pytest.mark.parametrize(
+    "estimator",
+    [
+        RandomForestClassifier(random_state=0),
+        make_pipeline(StandardScaler(), RandomForestClassifier(random_state=0)),
+    ],
+)
+def test_pandas_dataframe(estimator):
+    """If provided, the `mean_decrease_impurity` dataframe uses the feature names."""
+    X, y = make_classification(n_features=5, random_state=42)
+    X = pd.DataFrame(X, columns=[f"my_feature_{i}" for i in range(X.shape[1])])
+    estimator.fit(X, y)
+
+    report = EstimatorReport(estimator)
+    result = report.feature_importance.mean_decrease_impurity()
+
+    assert result.shape == (5, 1)
+    assert result.index.tolist() == [
+        "my_feature_0",
+        "my_feature_1",
+        "my_feature_2",
+        "my_feature_3",
+        "my_feature_4",
+    ]
+    assert result.columns.tolist() == ["Mean decrease impurity"]
+
+
+def _make_estimator_param(estimator):
+    return pytest.param(estimator, id=estimator.__class__.__name__)
+
+
+@pytest.mark.parametrize(
+    "estimator",
+    [
+        _make_estimator_param(estimator)
+        for estimator in [
+            sklearn.ensemble.AdaBoostClassifier(),
+            sklearn.ensemble.AdaBoostRegressor(),
+            sklearn.ensemble.ExtraTreesClassifier(),
+            sklearn.ensemble.ExtraTreesRegressor(),
+            sklearn.ensemble.GradientBoostingClassifier(),
+            sklearn.ensemble.GradientBoostingRegressor(),
+            sklearn.ensemble.RandomForestClassifier(),
+            sklearn.ensemble.RandomForestRegressor(),
+            sklearn.ensemble.RandomTreesEmbedding(),
+            sklearn.tree.DecisionTreeClassifier(),
+            sklearn.tree.DecisionTreeRegressor(),
+            sklearn.tree.ExtraTreeClassifier(),
+            sklearn.tree.ExtraTreeRegressor(),
```

Comment:
To keep the execution time quick since we are interested in testing the statistical soundness.

```suggestion
            sklearn.ensemble.AdaBoostClassifier(n_estimators=2),
            sklearn.ensemble.AdaBoostRegressor(n_estimators=2),
            sklearn.ensemble.ExtraTreesClassifier(n_estimators=2),
            sklearn.ensemble.ExtraTreesRegressor(n_estimators=2),
            sklearn.ensemble.GradientBoostingClassifier(n_estimators=2),
            sklearn.ensemble.GradientBoostingRegressor(n_estimators=2),
            sklearn.ensemble.RandomForestClassifier(n_estimators=2),
            sklearn.ensemble.RandomForestRegressor(n_estimators=2),
            sklearn.ensemble.RandomTreesEmbedding(n_estimators=2),
            sklearn.tree.DecisionTreeClassifier(),
            sklearn.tree.DecisionTreeRegressor(),
            sklearn.tree.ExtraTreeClassifier(),
            sklearn.tree.ExtraTreeRegressor(),
```